modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/CutMarksToSmilWorkflowOperationHandler.java

```diff
@@ -0,0 +1,319 @@
+
+  /**
+   * Callback for the OSGi declarative services configuration.
+   *
+   * @param composerService
+   *          the local composer service
+   */
+  public void setComposerService(ComposerService composerService) {
+    this.composerService = composerService;
+  }
+
+  /**
+   * Callback for declarative services configuration that will introduce us to the local workspace service.
+   * Implementation assumes that the reference is configured as being static.
+   *
+   * @param workspace
+   *          an instance of the workspace
+   */
+  public void setWorkspace(Workspace workspace) {
+    this.workspace = workspace;
+  }
+
+  /**
+   * The SMIL service to modify SMIL files.
+   */
+  private SmilService smilService;
+  public void setSmilService(SmilService smilService) {
+    this.smilService = smilService;
+  }
+
+  /** JSON Parser */
+  private static final Gson gson = new Gson();
+
+  /** Stores information read from JSON */
+  class Times {
+    private Long begin;
+    private Long duration;
+
+    public Long getStartTime() {
+      return begin;
+    }
+    public void setStartTime(Long startTime) {
+      this.begin = startTime;
+    }
+    public Long getDuration() {
+      return duration;
+    }
+    public void setDuration(Long duration) {
+      this.duration = duration;
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.workflow.api.WorkflowOperationHandler#start(org.opencastproject.workflow.api.WorkflowInstance,
+   *      JobContext)
+   */
+  @Override
+  public WorkflowOperationResult start(final WorkflowInstance workflowInstance, JobContext context)
+          throws WorkflowOperationException {
+    logger.info("Running cut marks to smil workflow operation on workflow {}", workflowInstance.getId());
+
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+    final MediaPackage mediaPackage = (MediaPackage) workflowInstance.getMediaPackage().clone();
+
+    // Read config options
+    final MediaPackageElementFlavor jsonFlavor = MediaPackageElementFlavor.parseFlavor(
+            getConfig(operation, SOURCE_JSON_FLAVOR));
+    final MediaPackageElementFlavor targetSmilFlavor = MediaPackageElementFlavor.parseFlavor(
+            getConfig(operation, TARGET_SMIL_FLAVOR));
+
+    String flavorNames = operation.getConfiguration(SOURCE_MEDIA_FLAVORS);
+    final List<MediaPackageElementFlavor> flavors = new ArrayList<MediaPackageElementFlavor>();
+    for (String flavorName : asList(flavorNames))
+      flavors.add(MediaPackageElementFlavor.parseFlavor(flavorName));
+
+    // Target tags
+    String targetTagsOption = StringUtils.trimToNull(operation.getConfiguration(TARGET_TAGS));
+    List<String> targetTags = asList(targetTagsOption);
+
+    // Is there a catalog?
+    Catalog[] catalogs = mediaPackage.getCatalogs(jsonFlavor);
+    if (catalogs.length != 1) {
+      logger.warn("Number of catalogs in the source flavor does not equal one. Skipping...");
```

Comment:
Skipping this if there is none may be fine but skipping it if there are multiple catalogs is weird and unexpected. I would expect that they are both converted or that there is an error.

Reply:
Right, that makes sense. I'll have it throw an error in that case for now, since processing multiple catalogs seems non-trivial (e.g. what if there are overlapping cutmarks).